sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/receiver.py

```diff
@@ -85,11 +85,13 @@ def _wait_for_response(self, wait: Union[bool, float]) -> None:
         if wait is True:
             self._session._connection.listen(wait=False) # pylint: disable=protected-access
             if self.state == LinkState.ERROR:
-                raise self._error
```

Comment:
dont we want to raise something here even if by some change self._error is none

Reply:
I think I can update the type of self._error in link.py so that I dont need this line https://github.com/Azure/azure-sdk-for-python/blob/940b433eb2ab950366fffa8cb3ad242599ee5856/sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/link.py#L233